core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -837,9 +840,13 @@ class Partition(val topicPartition: TopicPartition,
           val localLogInfo = (localBrokerId, localLogOrException.logEndOffset)
           val (ackedReplicas, awaitingReplicas) = (replicaInfo + localLogInfo).partition { _._2 >= requiredOffset}
 
-          trace(s"Progress awaiting ISR acks for offset $requiredOffset: " +
-            s"acked: ${ackedReplicas.map(logEndOffsetString)}, " +
-            s"awaiting ${awaitingReplicas.map(logEndOffsetString)}")
+          val timeElapsedMs = System.currentTimeMillis() - requestCreationTime
+          if (timeElapsedMs >= logSlowReplicationThresholdMs) {
```

Comment:
I feel like it would be worthwhile to add a counter (maybe associated with a per-minute or per-10-minute window to allow resets) and avoid logging if it's already happened too many times in the last N minutes or seconds). Basically you want a rate-limiter in case all requests are this slow for some reason (probably related to broker ill health, which you don't want to exacerbate).